src/GraphQL/DI/IGraphQLBuilder.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace GraphQL.DI
+{
+    /// <summary>
+    /// An interface for configuring GraphQL.NET services.
+    /// </summary>
+    public interface IGraphQLBuilder
+    {
+        /// <summary>
+        /// Registers the service of type <paramref name="serviceType"/> with the dependency injection provider.
```

Comment:
Worth to add some notes about double registration semantics. Something like
> Note that the last registration wins.

Reply:
Well, that’s only true if you don’t request an IEnumerable of the specific type. We could add a note to that effect I suppose.

Reply:
`IEnumerable<T>` is another type as such. You don't register `IEnumerable<T>` as is. It's just a marker to say DI "hey, give me ALL registrations of T". In any case, we need a comment describing the behavior with several registrations of the same T. All implementation should have this behavior. This is important, I have already encountered bugs with different semantics - https://github.com/EasyNetQ/EasyNetQ/pull/1108.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2769